project/src/Agents/GSTKinematicBody2DAgent.gd

```diff
@@ -15,30 +15,31 @@ var body: KinematicBody2D setget _set_body
 # SLIDE uses use move_and_slide
 # COLLIDE uses move_and_collide
 # POSITION changes the global_position directly
-var kinematic_movement_type: int = KinematicMovementType.SLIDE
+var movement_type: int
 
 var _last_position: Vector2
 
 
-func _init(body: KinematicBody2D) -> void:
+func _init(body: KinematicBody2D, movement_type: int = MovementType.SLIDE) -> void:
+	yield(body, "ready")
+	
 	self.body = body
-	if body.is_inside_tree():
-		body.get_tree().connect("physics_frame", self, "_on_SceneTree_frame")
-	else:
-		body.connect("ready", self, "_on_body_ready")
+	self.movement_type = movement_type
+	
+	body.get_tree().connect("physics_frame", self, "_on_SceneTree_physics_frame")
 
 
 # Moves the agent's `body` by target `acceleration`.
 # tags: virtual
-func _apply_steering(acceleration: GSTTargetAcceleration, delta: float) -> void:
+func apply_steering(acceleration: GSTTargetAcceleration, delta: float) -> void:
```

Comment:
It's a virtual function. By both [Godot's](https://docs.godotengine.org/en/3.2/getting_started/scripting/gdscript/gdscript_styleguide.html#functions-and-variables) and [our](https://www.gdquest.com/docs/guidelines/best-practices/godot-gdscript/#code-style) convention, it should have an underscore.